scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/authentication/ManageUsers.java

```diff
@@ -0,0 +1,506 @@
+            }
+            System.out.print("password: ");
+            if ((create && password == null) || (update && password == null && groups == null)) {
+                password = new String(console.readPassword());
+            } else {
+                System.out.println("*******");
+            }
+
+            if (create && groups == null) {
+                System.out.print("groups for user " + login + ":");
+                String groupString = console.readLine();
+                groups = groupString.split(",");
+            }
+        }
+
+        try {
+            Properties props = new Properties();
+            try {
+                FileInputStream stream = new FileInputStream(new File(loginFilePath));
+                props.load(stream);
+                stream.close();
+            } catch (Exception e) {
+                exitWithErrorMessage("could not read login file : " + loginFilePath, null, e);
+            }
+
+            Multimap<String, String> groupsMap = loadGroups(groupFilePath);
+
+            switch (action) {
+                case CREATE:
+                    if (props.containsKey(login)) {
+                        exitWithErrorMessage("USER already exists in login file : " + loginFilePath, null, null);
+                    }
+                    updateUserPassword(pubKey, login, password, props);
+                    System.out.println("Created user " + login + " in " + loginFilePath);
+                    updateUserGroups(login, groups, groupsMap);
+                    break;
+                case UPDATE:
+                    if (!props.containsKey(login)) {
+                        exitWithErrorMessage("USER does not exist in login file : " + loginFilePath, null, null);
+                    }
+                    if (password != null) {
+                        updateUserPassword(pubKey, login, password, props);
+                        System.out.println("Changed password for user " + login + " in " + loginFilePath);
+                    }
+                    if (groups != null) {
+                        updateUserGroups(login, groups, groupsMap);
+                    }
+                    break;
+                case DELETE:
+                    if (!props.containsKey(login)) {
+                        exitWithErrorMessage("USER does not exist in login file : " + loginFilePath, null, null);
+                    }
+                    props.remove(login);
+                    groupsMap.removeAll(login);
+                    System.out.println("Deleted user " + login + " in " + loginFilePath);
+                    break;
+            }
+            storeLoginFile(loginFilePath, props);
+
+            storeGroups(groupFilePath, groupsMap);
+        } catch (Throwable t) {
+            exitWithErrorMessage("Unexpected error", null, t);
+        }
+    }
+
+    private static void updateUserGroups(String login, String[] groups, Multimap<String, String> groupsMap) {
+        groupsMap.removeAll(login);
```

Comment:
yes in case a user for example had groups "admin, nsadmin" and we want to update it to "nsadmin". It's necessary to clean the map to remove previous groups of this user.